### PR TITLE
refactor: Simplify DocumentRepository

### PR DESCRIPTION
**Pull request by ArthurChalard**
This refactor deletes toMap and use built-in functions instead of visibilityFromSnapshot and visibilityToString

### CODE REVIEW

File: app/src/main/java/com/github/se/travelpouch/model/documents/DocumentContainer.kt [hunk 1]
```diff
@@ -61,12 +61,8 @@ enum class DocumentFileFormat(val mimeType: String) {
...
 
   companion object {
-    fun fromMimeType(mimeType: String?): DocumentFileFormat? {
-      for (value in values()) {
-        if (value.mimeType == mimeType) return value
-      }
-      return null
-    }
+    fun fromMimeType(mimeType: String?): DocumentFileFormat? =
+        values().find { it.mimeType == mimeType }
```
Comment (yassine04e) [inline app/src/main/java/com/github/se/travelpouch/model/documents/DocumentContainer.kt:65]: This is a good use of the find function.

**Review by yassine04e — COMMENTED** [review]
Are the function removed from DocumentRepository useless ?

**Comment by ArthurChalard** [comment]
Yes, indeed. Some were used, but they are now replaced by built-in functions

File: app/src/main/java/com/github/se/travelpouch/model/documents/DocumentRepository.kt [hunk 2]
```diff
@@ -202,24 +173,9 @@ class DocumentRepositoryFirestore(
           addedByEmail = document.getString("addedByEmail"),
           addedByUser = document.getDocumentReference("addedByUser"),
           addedAt = requireNotNull(document.getTimestamp("addedAt")),
-          visibility = requireNotNull(visibilityFromSnapshot(document)))
+          visibility = enumValueOf(requireNotNull(document.getString("visibility"))))
```
Comment (yassine04e) [inline app/src/main/java/com/github/se/travelpouch/model/documents/DocumentRepository.kt:176]: Yes, this is a good replacement

**Review by yassine04e — COMMENTED** [review]
I just have a final comment and then I think we can merge. Were the functions toMap(docuemnt : NewDocumentContainer) and visibilityToString not used ?

**Comment by ArthurChalard** [comment]
toMap was unused and visibilityToString was only used in toMap. Furthermore, the latest can be replaced by the built-in name.

**Review by yassine04e — APPROVED** [review]
This seems good to me now. I think we can merge

**Pull request merged**